libs/common-services/src/main/java/uk/gov/di/ipv/core/library/helpers/JwtHelper.java

```diff
@@ -58,4 +62,26 @@ public static boolean signatureIsDerFormat(SignedJWT signedJWT) throws JOSEExcep
         return signedJWT.getSignature().decode().length
                 != ECDSA.getSignatureByteArrayLength(JWSAlgorithm.ES256);
     }
+
+    private static String getKid(ConfigService configService) {
```

Comment:
It's annoying that you can't get this from the `JWSSigner` - it feels like an omission :(

Means that this must be kept in sync with the logic in `SignerFactory`, to make sure the key id actually matches the key being used in the signer!

Probably not the end of the world, but a potential improvement might be to either wrap in our own class (that does expose the key id) or combine this helper class with the signer factory, to have a single 'signing service' that handles the key selection in one place.